### PR TITLE
Add get transaction type method

### PR DESCRIPTION
**Pull request by nathanieliov**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

* **Other information**:

### CODE REVIEW

File: rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java
Comment (jeremy-then) [inline rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java:61]: Does this btc tx hex belong to the btc tx hash `99ff14d8d0b339f6144291940c5f99e8fd4002fb300cebde963555fd395ee943` above? I think that the security team advised us to avoid using real testnet or mainnet data in our test in another project. I'm not sure how it goes in this one.

**Review by jeremy-then — APPROVED** [review]
In general LGTM. Left a small comment.

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/PegUtils.java:24]: Is this the right exception to throw here? Seems specific to a certain method and this in theory a utils method that could be used by anyone

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (marcos-iov) [inline rskj-core/src/main/java/co/rsk/peg/PegUtils.java:27]: I think we discussed with Jose that we should first check if there is any value sent to any of the active federations, to avoid calculating the sighash (which may be an expensive operation) unnecessarily. If a random transaction is sent to the Bridge, we don't even bother in calculating the sighash. Which by the way we should a test for this case

Comment (marcos-iov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java:27]: Is there a difference between using `getWalletOutputs` and `getValueSentToMe`? I think I have seen both being used

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:42]: private?

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:80]: Any reason to send below the minimum? I think it kind of creates a situation where you are testing 2 things at once, below the minimum and unknown receiver

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:52]:
```suggestion
    @Test
```

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:36]: Transaction with multiple outputs to the active federation? Outputs to the active and retiring federation? Outputs to the federation and to other addresses? Sent from the retiring federation to the active but not in the index (weird case but should be treated as a peg-in)

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (kelvinator07) [inline rskj-core/src/main/java/co/rsk/peg/PegUtils.java:20]:
```suggestion
        Wallet liveFederationsWallet or activeFederationWallet,
```

File: rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/bitcoin/BitcoinUtilsTest.java:61]: Please see the changes related to the file in this PR: #2109. This PR is blocked by that one. It means that one needs to be merged before this.

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java:24]: Makes sense. I will return an `illegalStateException` then.

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java:27]: Regarding your first comment, I forgot to mention to you that if we use that order and assume that when there are no outputs to any live fed, it is an UNKNOWN type. Pegouts with no change output will be missed because those txs will be caught in that condition, considering those pegouts as UNKNOWN.

This is not a big problem because no utxo is added to the available utxo lists, but that pegout won't be marked as processed, and we will be storing a useless sigHash for those pegouts.

Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java:27]: Regarding your second comment, yes, we also use `getValueSentToMe`. It is used for validating peg-in and flyover peg-ins. Here, I used `getWalletOutputs` since we are only interested in knowing if exists outputs for any live fed. The validation of the amount will be done later.

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:42]: Done

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:80]: Since the previous method to identify the tx type also checks if pegin was valid, I wanted to prove this method doesn't care about the amount of the tx.

Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:80]: No reason. Fixed.

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:52]: Done

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java:22]: Maybe. But this out of scope for ticket.

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java:20]: By live federations, I mean retiring and active federations. This term is widely used in this project as well as others like fed-changer.

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:36]: Done. Migration txs but not in the index were there already.

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (kelvinator07) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java:22]: Then It Can Be Updated To BridgeIllegalArgumentException, RegisterBtcTransactionException Is Kinda Specific

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java]: Done

File: rskj-core/src/main/java/co/rsk/peg/PegUtils.java
Comment (nathanieliov) [reply rskj-core/src/main/java/co/rsk/peg/PegUtils.java]: Changed it to `IllegalStateException`.

**Review by kelvinator07 — APPROVED** [review]

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (kelvinator07) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:40]:
```suggestion
    private ActivationConfig.ForBlock activations = ActivationConfigsForTest.tbd600().forBlock(0);
```

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (kelvinator07) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:95]: Do We Need The Pegin/Pegout Amount To Determine The Transaction Type?

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (marcos-iov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:123]: Same for active and retired federations. You could have them both as global variables. Retired = genesis, Active = P2shEprFed

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:95]: The legacy method does, but since this is one that only aims to identify the type without checking if tx is valid, I wanted to assert the amount is not being used anymore to check pegins.

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:207]: Done

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java]: Update

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java]: Updated

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java]: Done

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java]: Updated

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java:74]: I think we should set this up for each test. Not sure it's the best idea to have active and retiring federation always by default

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java
Comment (marcos-iov) [inline rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java:285]: Could we pass a mock for activations object here and avoid receiving it as param? It doesn't change anything for P2SH ERP federations

File: rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegUtilsTest.java]: Done

File: rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java
Comment (nathanieliov) [reply rskj-core/src/test/java/co/rsk/peg/PegTestUtils.java]: Done

**Review by marcos-iov — APPROVED** [review]

**Pull request merged**
